packages/@aws-cdk/aws-lambda-event-sources/lib/dynamodb.ts

```diff
@@ -22,10 +24,18 @@ export class DynamoEventSource extends StreamEventSource {
       throw new Error(`DynamoDB Streams must be enabled on the table ${this.table.node.path}`);
     }
 
-    target.addEventSourceMapping(`DynamoDBEventSource:${this.table.node.uniqueId}`,
+    const eventSourceMapping = target.addEventSourceMapping(`DynamoDBEventSource:${this.table.node.uniqueId}`,
       this.enrichMappingOptions({eventSourceArn: this.table.tableStreamArn})
     );
+    this._eventSourceMappingId = eventSourceMapping.eventSourceMappingId;
 
     this.table.grantStreamRead(target);
   }
+
+  /**
+   * The Ref of the EventSourceMapping
+   */
+  public get eventSourceMappingId(): string | undefined {
+    return this._eventSourceMappingId;
+  }
```

Comment:
I would suggest that we throw an `Error` here instead of returning `undefined`, with the message 'DynamoEventSource is not yet bound to an event source mapping'.